livekit-plugins/livekit-plugins-sarvam/livekit/plugins/sarvam/tts.py

```diff
@@ -0,0 +1,274 @@
+    base_url: str = SARVAM_TTS_BASE_URL
+
+
+class TTS(tts.TTS):
+    """Sarvam.ai Text-to-Speech implementation.
+
+    This class provides text-to-speech functionality using the Sarvam.ai API.
+    Sarvam.ai specializes in high-quality TTS for Indian languages.
+
+    Args:
+        target_language_code: BCP-47 language code for supported Indian languages
+        model: Sarvam TTS model to use (only bulbul:v2 supported)
+        speaker: Voice to use for synthesis
+        speech_sample_rate: Audio sample rate in Hz
+        num_channels: Number of audio channels (Sarvam outputs mono)
+        pitch: Voice pitch adjustment (-20.0 to 20.0)
+        pace: Speech rate multiplier (0.5 to 2.0)
+        loudness: Volume multiplier (0.5 to 2.0)
+        enable_preprocessing: Whether to use text preprocessing
+        api_key: Sarvam.ai API key (required)
+        base_url: API endpoint URL
+        http_session: Optional aiohttp session to use
+    """
+
+    def __init__(
+        self,
+        *,
+        target_language_code: SarvamTTSLanguages | str,
+        model: SarvamTTSModels | str = "bulbul:v2",
+        speaker: SarvamTTSSpeakers | str = "manisha",
+        speech_sample_rate: int = 22050,
+        num_channels: int = 1,  # Sarvam output is mono WAV
+        pitch: float = 0.0,
+        pace: float = 1.0,
+        loudness: float = 1.0,
+        enable_preprocessing: bool = False,
+        api_key: str,
+        base_url: str = SARVAM_TTS_BASE_URL,
+        http_session: aiohttp.ClientSession | None = None,
+    ) -> None:
+        super().__init__(
+            capabilities=tts.TTSCapabilities(streaming=False),
+            sample_rate=speech_sample_rate,
+            num_channels=num_channels,
+        )
+
+        self._api_key = api_key or os.environ.get("SARVAM_API_KEY")
+        if not self._api_key:
+            raise ValueError(
+                "Sarvam API key is required. Provide it directly or set SARVAM_API_KEY env var."
+            )
+
+        # Validate model-speaker compatibility
+        if not validate_model_speaker_compatibility(model, speaker):
+            compatible_speakers = MODEL_SPEAKER_COMPATIBILITY.get(model, {}).get("all", [])
+            raise ValueError(
+                f"Speaker '{speaker}' is not compatible with model '{model}'. "
+                f"Please choose a compatible speaker from: {', '.join(compatible_speakers)}"
+            )
+
+        self._opts = SarvamTTSOptions(
+            target_language_code=target_language_code,
+            model=model,
+            speaker=speaker,
+            speech_sample_rate=speech_sample_rate,
+            pitch=pitch,
+            pace=pace,
+            loudness=loudness,
+            enable_preprocessing=enable_preprocessing,
+            api_key=self._api_key,
+            base_url=base_url,
+        )
+        self._session = http_session
+        self._logger = logger.getChild(self.__class__.__name__)
```

Comment:
We don't use logger this way, we should just import the logger from the log.py file...